syntax = "proto3";

package master_slave;

service MasterSlaveService {
  rpc RegisterSlave (SlaveInfo) returns (RegistrationResponse);
  rpc SendResult (Result) returns (Acknowledgement);
}

message SlaveInfo {
  string slave_id = 1;
  string ip = 2;
  string timestamp = 3;
}

message RegistrationResponse {
  string message = 1;
}

message Result {
  string slave_id = 1;
  string data = 2;
}

message Acknowledgement {
  string message = 1;
}